File: frontend-web/lib/auth/token-storage.ts
const ACCESS_TOKEN_KEY = "menugreen_access_token";
const REFRESH_TOKEN_KEY = "menugreen_refresh_token";
const FULL_NAME_KEY = "menugreen_full_name";

function isBrowser(): boolean {
  return typeof window !== "undefined";
}

export const tokenStorage = {
  getAccessToken(): string | null {
    if (!isBrowser()) return null;
    return localStorage.getItem(ACCESS_TOKEN_KEY);
  },

  getRefreshToken(): string | null {
    if (!isBrowser()) return null;
    return localStorage.getItem(REFRESH_TOKEN_KEY);
  },

  getFullName(): string | null {
    if (!isBrowser()) return null;
    return localStorage.getItem(FULL_NAME_KEY);
  },

  saveTokens(input: {
    accessToken: string;
    refreshToken: string;
    fullName?: string | null;
  }): void {
    if (!isBrowser()) return;
    localStorage.setItem(ACCESS_TOKEN_KEY, input.accessToken);
    localStorage.setItem(REFRESH_TOKEN_KEY, input.refreshToken);
    if (input.fullName) {
      localStorage.setItem(FULL_NAME_KEY, input.fullName);
    }
  },

  clear(): void {
    if (!isBrowser()) return;
    localStorage.removeItem(ACCESS_TOKEN_KEY);
    localStorage.removeItem(REFRESH_TOKEN_KEY);
    localStorage.removeItem(FULL_NAME_KEY);
  },
};

File: frontend/lib/core/network/token_storage.dart
import 'package:shared_preferences/shared_preferences.dart';

class TokenStorage {
  static const _accessTokenKey = 'access_token';
  static const _refreshTokenKey = 'refresh_token';
  static const _fullNameKey = 'full_name';
  static const _userIdKey = 'user_id';

  Future<String?> getAccessToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_accessTokenKey);
  }

  Future<String?> getRefreshToken() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_refreshTokenKey);
  }

  Future<String?> getFullName() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_fullNameKey);
  }

  Future<String?> getUserId() async {
    final prefs = await SharedPreferences.getInstance();
    return prefs.getString(_userIdKey);
  }

  Future<void> saveUserId(String userId) async {
    if (userId.isEmpty) return;
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_userIdKey, userId);
  }

  Future<void> saveFullName(String fullName) async {
    if (fullName.isEmpty) return;
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_fullNameKey, fullName);
  }

  Future<void> saveTokens({
    required String accessToken,
    required String refreshToken,
    String? fullName,
    String? userId,
  }) async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(_accessTokenKey, accessToken);
    await prefs.setString(_refreshTokenKey, refreshToken);
    if (fullName != null && fullName.isNotEmpty) {
      await prefs.setString(_fullNameKey, fullName);
    }
    if (userId != null && userId.isNotEmpty) {
      await prefs.setString(_userIdKey, userId);
    }
  }

  Future<void> clear() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.remove(_accessTokenKey);
    await prefs.remove(_refreshTokenKey);
    await prefs.remove(_fullNameKey);
    await prefs.remove(_userIdKey);
  }
}

